service/resource_router.go

```diff
@@ -0,0 +1,33 @@
+package service
+
+import (
+	"context"
+
+	"github.com/giantswarm/kvm-operator/service/keyv2"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+// newResourceRouter determines which resources are enabled based upon the
+// version in the version bundle.
+func newResourceRouter(resources []framework.Resource) func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
```

Comment:
In AWS we inject a map indexed by the version bundle version being supported. Routing could be as easy as this. 

```
resourceList, ok := resourceMap[versionBundleVersion]
if ok {
  return resourceList, nil
} 

return nil, microerror.Maskf(...)
```

We are going to need to have it that way anyway, so I would like to do it that way right away.